src/controllers/ajax/IndexAjaxController.php

```diff
@@ -119,6 +131,25 @@ protected function getActions(): array<string> {
 
     // Check logo
     $final_logo = $logo;
+
+    if ($is_custom_logo) {
+      $logo_data = str_replace(' ', '+', $final_logo);
+      $logo_data_unbase64ed = base64_decode($logo_data);
+
+      $name = 'custom-'.time();
+      $filename = $name.'.'.$logo_type;
+      // '..' is separated for DB storage
+      $url = self::CUSTOM_LOGO_DIR.$filename;
+      $filepath = dirname(__FILE__).'/../..'.$url;
```

Comment:
I would prefer avoiding the "../../" type of expressions. The path for custom logos is always the same, and since you are generating the filename, just use them statically.